src/Core/Localization/CLDR/Reader.php

```diff
@@ -205,12 +205,7 @@ protected function getParentLocale($localeCode)
         // The common case with truncation
         $pos = strrpos($localeCode, '_');
         if (false !== $pos) {
-            $parent = substr($localeCode, 0, $pos);
-            if (false === $parent) {
-                throw new LocalizationException(sprintf('Invalid locale code: "%s"', $localeCode));
-            }
-
-            return $parent;
+            return substr($localeCode, 0, $pos);
```

Comment:
Only with PHP8: 

8.0.0 | The function returns an empty where it previously returned false.
-- | --



Keep the original code please

Reply:
`strrpos` returns `int|false`.

`false` is filtered at l. 207.

Concerning the result of `substr`, the return is a string.
> If length is given and is positive, the string returned will contain at most length characters beginning from offset (depending on the length of string). 
> If length is given and is 0, false or null, an empty string will be returned.

`false` can't be returned in this case : length is not negative
> If length is given and is negative, then that many characters will be omitted from the end of string (after the start position has been calculated when a offset is negative). If offset denotes the position of this truncation or beyond, false will be returned. 

`false` can't be returned in this case : `$pos` is a return of `strrpos`.
> If string is less than offset characters long, false will be returned